include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+                             "_symmetric_gradient_q" +
+                             Utilities::int_to_string(n_q_points) + "_" +
+                             Utilities::type_to_string(dummy);
+
+    // Now build the return type
+    using RetType =
+      std::vector<typename OutputType<dim, spacedim, Number, Extractor>::
+                    symmetric_gradient_type>;
+
+
+    if (!stores_object_with_name(name))
+      add_copy(RetType(n_q_points), name);
+
+    RetType &ret = get_object_with_name<RetType>(name);
+
+    AssertDimension(ret.size(), n_q_points);
+
+    fev[variable].get_function_symmetric_gradients_from_local_dof_values(
+      independent_local_dofs, ret);
+    return ret;
+  }
+
+
+  template <int dim, int spacedim>
+  template <typename Extractor, typename Number>
+  const std::vector<
+    typename OutputType<dim, spacedim, Number, Extractor>::divergence_type> &
+  ScratchData<dim, spacedim>::get_divergences(
+    const std::string &global_vector_name,
+    const Extractor &  variable,
+    const Number       dummy)
+  {
+    const std::vector<Number> &independent_local_dofs =
+      get_local_dof_values(global_vector_name, dummy);
+
+    const FEValuesBase<dim, spacedim> &fev = get_current_fe_values();
+
+    const unsigned int n_q_points = fev.n_quadrature_points;
+
+    const std::string name = global_vector_name + "_" + variable.get_name() +
+                             "_divergence_q" +
+                             Utilities::int_to_string(n_q_points) + "_" +
+                             Utilities::type_to_string(dummy);
+
+    // Now build the return type
+    using RetType = std::vector<
+      typename OutputType<dim, spacedim, Number, Extractor>::divergence_type>;
+
+
+    if (!stores_object_with_name(name))
+      add_copy(RetType(n_q_points), name);
+
+    RetType &ret = get_object_with_name<RetType>(name);
+
+    AssertDimension(ret.size(), n_q_points);
+
+    fev[variable].get_function_divergences_from_local_dof_values(
+      independent_local_dofs, ret);
+    return ret;
+  }
+
+
+
+  template <int dim, int spacedim>
+  template <typename Extractor, typename Number>
+  const std::vector<
+    typename OutputType<dim, spacedim, Number, Extractor>::curl_type> &
+  ScratchData<dim, spacedim>::get_curls(const std::string &global_vector_name,
+                                        const Extractor &  variable,
+                                        const Number       dummy)
+  {
+    const std::vector<Number> &independent_local_dofs =
+      get_local_dof_values(global_vector_name, dummy);
+
+    const FEValuesBase<dim, spacedim> &fev = get_current_fe_values();
+
+    const unsigned int n_q_points = fev.n_quadrature_points;
+
+    const std::string name = global_vector_name + "_" + variable.get_name() +
+                             "_curl_q" + Utilities::int_to_string(n_q_points) +
+                             "_" + Utilities::type_to_string(dummy);
```

Comment:
I think that here you're trying to create a unique string that a user is unlikely going to step on. How about you start it with something very unlikely, say `_____` or `\0` or similar? Not that it's going to matter much, I suspect :-) 

Since you do this repeatedly in the functions above, you might want to write a small helper function that builds this kind of string in one place, rather than repeating the setup in multiple places.

Reply:
Now this is no longer possible, since there are two DataStorage objects: one for internal use, and one for the users. The internal one is only accessible by us, so we can do whatever we want with the names.

As far as the helper function, its signature would be almost as complicated as writing this out, with the disadvantage that you'd have to look up the other function to actually figure out what the name is...
~~~
const std::name = get_unique_name(global_vector_name, variable, "curl_q", n_q_points, dummy);
~~~